templates/markdown/type.tpl

```diff
@@ -31,7 +31,7 @@ _Appears in:_
 {{ end -}}
 
 {{ range $type.Members -}}
-| `{{ .Name  }}` _{{ markdownRenderType .Type }}_ | {{ template "type_members" . }} | {{ markdownRenderDefault .Default }} | {{ range .Validation -}} {{ . }} <br />{{ end }} |
+| `{{ .Name  }}` _{{ markdownRenderType .Type }}_ | {{ template "type_members" . }} | {{ markdownRenderDefault .Default }} | {{ range .Validation -}} {{ markdownRenderFieldDoc . }} <br />{{ end }} |
```

Comment:
So we reuse `markdownRenderFieldDoc` to render the custom validation marker. It's ok-ish but not very clean for me. Similar to https://github.com/elastic/crd-ref-docs/pull/81#discussion_r1686764298, let's merge and refactor this later.